$\min \ -3x_1-2x_2+8x_3-7x_4+0.19x_5+11x_6+20.2x_7 \\ \text{ s.t. }  \sum_{1}^{7}-x_i \geq -256 \\  \sum_{1}^{7}x_i \geq 256 \\ -x_2+x_3 \geq 10 \\ 3x1-x_5+12x_6 \geq -105 \\ 18.5x_2+30.5x_4-x_7 \geq 28 \\ -x_3-4x_5+x_7 \geq 31 \\ x_1 \geq -5,x_2 \geq0,x_3 \geq 5,x_4 \geq -3,x_5 \geq -4, \\x_6 \geq 0,x_7 \geq 10 $.  


In [ ]:
!pip install -q pyomo


In [ ]:
from pyomo.environ import *

In [ ]:
# create a model1
model1 = ConcreteModel()

In [ ]:
# declare decision variables
model1.x1 = Var(domain=Reals)
model1.x2 = Var(domain=Reals)
model1.x3 = Var(domain=Reals)
model1.x4 = Var(domain=Reals)
model1.x5= Var(domain=Reals)
model1.x6 = Var(domain=Reals)
model1.x7 = Var(domain=Reals)


In [ ]:
# declare objective
model1.cost = Objective(expr = -3*model1.x1 - 2*model1.x2 + 8*model1.x3 -7*model1.x4+.19*model1.x5+11*model1.x6+20.2*model1.x7, sense=minimize)

In [ ]:
# declare constraints
model1.constr1 = Constraint(expr = model1.x1+model1.x2+model1.x3+model1.x4+model1.x5+model1.x6 +model1.x7==256)
model1.constr2 = Constraint(expr = -1*model1.x2 +model1.x3>= 10)

model1.constr3 = Constraint(expr = -3*model1.x1 + model1.x5-12*model1.x6 <= 105)
model1.constr4 = Constraint(expr = 18.5*model1.x1 + 30.5*model1.x4 -model1.x7  >= 28)
model1.constr5 = Constraint(expr = -1*model1.x3 -4*model1.x5+model1.x7>= 31)
model1.constr6 = Constraint(expr = model1.x1>= -5)
model1.constr7 = Constraint(expr = model1.x2>=0)
model1.constr8 = Constraint(expr = model1.x3>= 5)
model1.constr9= Constraint(expr = model1.x4>= -3)
model1.constr10 = Constraint(expr = model1.x5>= -4)
model1.constr11= Constraint(expr = model1.x6>= 0)
model1.constr12= Constraint(expr = model1.x7>= 10)


In [ ]:
model1.pprint()

7 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x6 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x7 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fi

In [ ]:
!apt-get install -y -qq glpk-utils

Selecting previously unselected package libsuitesparseconfig5:amd64.
(Reading database ... 155673 files and directories currently installed.)
Preparing to unpack .../libsuitesparseconfig5_1%3a5.1.2-2_amd64.deb ...
Unpacking libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libamd2:amd64.
Preparing to unpack .../libamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libcolamd2:amd64.
Preparing to unpack .../libcolamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libcolamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libglpk40:amd64.
Preparing to unpack .../libglpk40_4.65-1_amd64.deb ...
Unpacking libglpk40:amd64 (4.65-1) ...
Selecting previously unselected package glpk-utils.
Preparing to unpack .../glpk-utils_4.65-1_amd64.deb ...
Unpacking glpk-utils (4.65-1) ...
Setting up libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Setting up libcolamd2:amd64 (1:5.1.2-2) ...
Setting up libamd2:amd64 

In [ ]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model1).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: -1010.76
  Upper bound: -1010.76
  Number of objectives: 1
  Number of constraints: 13
  Number of variables: 8
  Number of nonzeros: 26
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.011832475662231445
# ----------------------------------------------------------
#   Solution Information
# ----------------------------

In [ ]:
# display solution
print('\nCost = ', model1.cost())

print('\nDecision Variables')
print('x1 = ', model1.x1())
print('x2 = ', model1.x2())
print('x3 = ', model1.x3())
print('x4 = ', model1.x4())
print('x5 = ', model1.x5())
print('x6 = ', model1.x6())
print('x7 = ', model1.x7())



print('\nConstraints')
print('Constraint 1  = ', model1.constr1())
print('Constraint 2  = ', model1.constr2())
print('Constraint 3  = ', model1.constr3())
print('Constraint 4  = ', model1.constr4())
print('Constraint 5  = ', model1.constr5())
print('Constraint 6  = ', model1.constr6())
print('Constraint 7  = ', model1.constr7())
print('Constraint 8  = ', model1.constr8())
print('Constraint 9  = ', model1.constr9())
print('Constraint 10  = ', model1.constr10())
print('Constraint 11 = ', model1.constr11())
print('Constraint 12 = ', model1.constr12())



Cost =  -1010.76

Decision Variables
x1 =  -5.0
x2 =  0.0
x3 =  10.0
x4 =  230.0
x5 =  -4.0
x6 =  0.0
x7 =  25.0

Constraints
Constraint 1  =  256.0
Constraint 2  =  10.0
Constraint 3  =  11.0
Constraint 4  =  6897.5
Constraint 5  =  31.0
Constraint 6  =  -5.0
Constraint 7  =  0.0
Constraint 8  =  10.0
Constraint 9  =  230.0
Constraint 10  =  -4.0
Constraint 11 =  0.0
Constraint 12 =  25.0


Ans:
the objettive value are :-1010.7600\\
 value of the  decision variables are  and optimal point are: \\
x1 =  -5.0, \\
x2 =  0.0, \\
x3 =  10.0, \\
x4 =  230.0, \\
x5 =  -4.0, \\
x6 =  0.0, \\
x7 =  24.999 \\
Constraints activities are= \\
 constraint 1  active because LHS = $ = -256 = RHS $ \\ , 
 constraint 2  active because LHS = $ = 256 = RHS $ \\
 constraint 3 are active because LHS = $  = 10 = RHS $ \\
 constraint 4 are  inactive since LHS = are non equal to rhs\\
constraint 5 are  inactive since LHS = are non equal to rhs